tiledb/api/src/array/schema/mod.rs

```diff
@@ -21,13 +21,22 @@ use crate::Datatype;
 use crate::{Factory, Result as TileDBResult};
 
 #[derive(
-    Clone, Copy, Debug, Deserialize, Eq, OptionSubset, PartialEq, Serialize,
+    Clone,
+    Default,
+    Copy,
+    Debug,
+    Deserialize,
+    Eq,
+    OptionSubset,
+    PartialEq,
+    Serialize,
 )]
 #[cfg_attr(
     any(test, feature = "proptest-strategies"),
     derive(proptest_derive::Arbitrary)
 )]
 pub enum ArrayType {
+    #[default]
```

Comment:
I don't love this, as you probably know...
so the goal is really that we want to be able to do
```
SchemaData {
    /* other fields */
    ..Default::default()
}
```
i.e. do the initializer list for the fields we want to supply values for, with `Default::default()` to get the ones we don't care about.  And this piece is the chosen way to implement that.

This is kind of what the Builder pattern is for. `Builder::new` requires supplying the non-default args, and then everything else is a setter that returns `self`, so you end up with something that looks like of like the literal constructor.

Reply:
What do you think of adding `Invalid` as a variant that blows up as soon as we try and convert it? For datatypes I think we decided that the same behavior by using `Any` as the default would be acceptable? Granted this adds a new variant that has no meaning in C++ land, though, I'm not sure that concerns me.

Reply:
That smells like an Option but without the strong semantics, doesn't it? Might as well make all the `ThingData` classes have `Option<ArrayType>`.

Except not quite, because you do get the relative user-friendliness of not having to write anything down for the other fields, as you show here...

I think I like the constructor and builder the most but I like having `Dense` as the default more than I like having an `Option`, so I'll accept this change.